code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -596,8 +596,12 @@ to_chat will check for valid clients itself already so no need to double check f
 		chosen_silo = input("Available Egg Silos") as null|anything in possible_silos
 		xeno_candidate.forceMove(chosen_silo)
 		var/double_check = input(xeno_candidate, "Spawn here?", "Spawn location") as null|anything in list("Yes","No")
-		if(double_check != "Yes")
+		if(double_check == "No")
 			attempt_to_spawn_larva_in_silo(xeno_candidate, possible_silos)
+		else if(double_check == "Yes")
+			//just continue on below
+		else
```

Comment:
This is band and you know it

Reply:
Would you rather a switch()?

Reply:
Not a switch, the empty elseif clause

Reply:
Then may I please get an actual suggestion because it is working exactly as I want it to with this code.

Reply:
Else If isnot yes

Because considering you're dodging a compiler warning on purpose It should be obvious something is wrong

Reply:
well there are three possible states here.

"No" loop again to choose another silo
"Yes" I have chosen the correct silo and should attempt to spawn
`NULL` I don't want to spawn anymore.

with `if(!"Yes")` how can it decide between nothing and no.

Reply:
if no
else if yes
else

???